multiapps-controller-persistence/src/main/java/org/cloudfoundry/multiapps/controller/persistence/util/SqlQueryExecutor.java

```diff
@@ -41,14 +41,22 @@ public <R> R execute(SqlQuery<R> sqlQuery) throws SQLException {
             JdbcUtil.rollback(connection);
             throw e;
         } finally {
-            setAutocommitSafely(connection);
+            JdbcUtil.setAutoCommitSafely(connection);
             JdbcUtil.closeQuietly(connection);
         }
     }
 
-    private void setAutocommitSafely(Connection connection) throws SQLException {
-        if (connection != null) {
-            connection.setAutoCommit(true);
+    public <R> R executeWithoutCommit(SqlQuery<R> sqlQuery) throws SQLException {
+        Connection connection = null;
+        try {
+            connection = dataSource.getConnection();
+            connection.setAutoCommit(false);
+            return sqlQuery.execute(connection);
+        } catch (SQLException e) {
+            JdbcUtil.rollback(connection);
+            JdbcUtil.setAutoCommitSafely(connection);
```

Comment:
what is the point of setting the autoCommit here?

Reply:
just in case, to make sure we tried every way to commit/rollback the transaction